diff --git a/vimode.py b/vimode.py
index 06f9c28..f759ac7 100755
--- a/vimode.py
+++ b/vimode.py
@@ -1831,6 +1831,10 @@ def check_warnings():
 # ============
 
 if __name__ == "__main__":
+    VI_DEFAULT_KEYS['\\'] = VI_KEYS[','] = key_comma
+    del VI_DEFAULT_KEYS[',']
+    del VI_KEYS[',']
+
     weechat.register(SCRIPT_NAME, SCRIPT_AUTHOR, SCRIPT_VERSION,
                      SCRIPT_LICENSE, SCRIPT_DESC, "", "")
     # Warn the user if he's using an unsupported WeeChat version.
