diff --git a/notify_send.py b/notify_send.py
index 6c10a5c..1fb65ac 100644
--- a/notify_send.py
+++ b/notify_send.py
@@ -27,6 +27,7 @@
 # SOFTWARE.
 #
 
+import getpass
 import os
 import subprocess
 import sys
@@ -283,12 +284,48 @@ def notification_should_be_sent_disregarding_time(buffer, tags, nick,
     if is_highlight:
         return notify_on_highlights()
 
+    if is_buddy(nick):
+        return True
+
     if notify_on_all_messages_in_buffer(buffer):
         return True
 
     return False
 
 
+def is_buddy(nick):
+    buddylist = getattr(is_buddy, 'buddylist', None)
+    count = getattr(is_buddy, 'count', None)
+
+    predicates = [
+        buddylist is None,
+        count is None,
+    ]
+
+    if count is not None:
+        predicates.append(count >= 100)
+
+    load_buddylist = any(predicates)
+
+    if load_buddylist:
+        is_buddy.buddylist = []
+        is_buddy.count = 1
+
+        for line in open('/home/{}/.weechat/buddylist.txt'.format(getpass.getuser()), 'r'):
+            buddy = line.split('.', maxsplit=1)[-1].strip()
+            is_buddy.buddylist.append(buddy)
+        buddylist = is_buddy.buddylist
+        with open('/var/tmp/notify_send.log', 'a+') as f:
+            f.write('BUDDY LIST:: {}\n\n'.format(buddylist))
+    else:
+        is_buddy.count += 1
+
+    if nick in buddylist:
+        return True
+    else:
+        return False
+
+
 def is_below_min_notification_delay(buffer):
     """Is a notification in the given buffer below the minimal delay between
     successive notifications from the same buffer?
