diff --git a/notify_send.py b/notify_send.py
index 6c10a5c..ccd1068 100644
--- a/notify_send.py
+++ b/notify_send.py
@@ -229,6 +229,10 @@ def message_printed_callback(data, buffer, date, tags, is_displayed,
     nick = nick_that_sent_message(tags, prefix)
 
     if notification_should_be_sent(buffer, tags, nick, is_displayed, is_highlight):
+        if is_highlight:
+            with open('/home/bryan/.weechat/highlights.txt', 'a+') as f:
+                f.write('[#{}] ({}) {}: {}'.format(buffer, date, nick, message))
+
         notification = prepare_notification(buffer, nick, message)
         send_notification(notification)
 
