diff --git a/notify_send.py b/notify_send.py
index 6349da1..90ea8bc 100644
--- a/notify_send.py
+++ b/notify_send.py
@@ -236,6 +236,16 @@ def message_printed_callback(data, buffer, date, tags, is_displayed,
     nick = nick_that_sent_message(tags, prefix)
 
     if notification_should_be_sent(buffer, tags, nick, is_displayed, is_highlight, message):
+        # log notification
+        import datetime as dt
+        with open('/home/bryan/.weechat/notifications.log', 'a+') as f:
+            now = dt.datetime.now()
+            fmt = '[{}] ({}) @{}: {}\n'
+            f.write(fmt.format(now.strftime('%Y-%m-%d %H:%M:%S'),
+                               weechat.buffer_get_string(buffer, 'short_name'),
+                               nick,
+                               message))
+
         notification = prepare_notification(buffer, nick, message)
         send_notification(notification)
 
