diff --git a/include/wkhtmltox/multipageloader.hh b/include/wkhtmltox/multipageloader.hh
index 42191ee..e5b7639 100644
--- a/include/wkhtmltox/multipageloader.hh
+++ b/include/wkhtmltox/multipageloader.hh
@@ -24,7 +24,7 @@
 #if QT_VERSION >= 0x050000
 #include <QtWebKitWidgets>
 #else
-#include <QWebPage>
+#include <qwebpage.h>
 #endif
 #include <wkhtmltox/loadsettings.hh>
 
diff --git a/src/image/imagedocparts.cc b/src/image/imagedocparts.cc
index 1272746..fcdf184 100644
--- a/src/image/imagedocparts.cc
+++ b/src/image/imagedocparts.cc
@@ -20,7 +20,7 @@
 
 #include "imagecommandlineparser.hh"
 #include "outputter.hh"
-#include <QWebFrame>
+#include <qwebframe.h>
 
 #define STRINGIZE_(x) #x
 #define STRINGIZE(x) STRINGIZE_(x)
diff --git a/src/image/wkhtmltoimage.cc b/src/image/wkhtmltoimage.cc
index 2977ffa..c32ab42 100644
--- a/src/image/wkhtmltoimage.cc
+++ b/src/image/wkhtmltoimage.cc
@@ -21,7 +21,7 @@
 #include "imagecommandlineparser.hh"
 #include "progressfeedback.hh"
 #include <QApplication>
-#include <QWebFrame>
+#include <qwebframe.h>
 #include <wkhtmltox/imageconverter.hh>
 #include <wkhtmltox/imagesettings.hh>
 #include <wkhtmltox/utilities.hh>
diff --git a/src/lib/converter.cc b/src/lib/converter.cc
index 0111504..11c7d37 100644
--- a/src/lib/converter.cc
+++ b/src/lib/converter.cc
@@ -21,7 +21,7 @@
 
 #include "converter_p.hh"
 #include "multipageloader.hh"
-#include <QWebFrame>
+#include <qwebframe.h>
 #include <qapplication.h>
 
 #ifdef QT4_STATICPLUGIN_TEXTCODECS
diff --git a/src/lib/converter_p.hh b/src/lib/converter_p.hh
index a55866a..bb1a872 100644
--- a/src/lib/converter_p.hh
+++ b/src/lib/converter_p.hh
@@ -24,7 +24,7 @@
 #include "converter.hh"
 #include "websettings.hh"
 #include <QFile>
-#include <QWebSettings>
+#include <qwebsettings.h>
 
 #include "dllbegin.inc"
 
diff --git a/src/lib/imageconverter.cc b/src/lib/imageconverter.cc
index 09e3b23..03f1b47 100644
--- a/src/lib/imageconverter.cc
+++ b/src/lib/imageconverter.cc
@@ -31,9 +31,9 @@
 #include <QPainter>
 #include <QSvgGenerator>
 #include <QUrl>
-#include <QWebElement>
-#include <QWebFrame>
-#include <QWebPage>
+#include <qwebelement.h>
+#include <qwebframe.h>
+#include <qwebpage.h>
 #include <qapplication.h>
 
 #ifdef Q_OS_WIN32
diff --git a/src/lib/multipageloader.hh b/src/lib/multipageloader.hh
index 9874c9a..bfbd84e 100644
--- a/src/lib/multipageloader.hh
+++ b/src/lib/multipageloader.hh
@@ -27,7 +27,7 @@
 #if QT_VERSION >= 0x050000
 #include <QtWebKitWidgets>
 #else
-#include <QWebPage>
+#include <qwebpage.h>
 #endif
 #include <wkhtmltox/loadsettings.hh>
 
diff --git a/src/lib/multipageloader_p.hh b/src/lib/multipageloader_p.hh
index 8a1939c..e155ad7 100644
--- a/src/lib/multipageloader_p.hh
+++ b/src/lib/multipageloader_p.hh
@@ -30,7 +30,7 @@
 #include <QNetworkAccessManager>
 #include <QNetworkCookieJar>
 #include <QNetworkReply>
-#include <QWebFrame>
+#include <qwebframe.h>
 
 #include "dllbegin.inc"
 namespace wkhtmltopdf {
diff --git a/src/lib/outline.hh b/src/lib/outline.hh
index edb825a..fb7c470 100644
--- a/src/lib/outline.hh
+++ b/src/lib/outline.hh
@@ -22,8 +22,8 @@
 #define __OUTLINE_HH__
 
 #include "pdfsettings.hh"
-#include <QWebElement>
-#include <QWebFrame>
+#include <qwebelement.h>
+#include <qwebframe.h>
 
 #ifdef  __EXTENSIVE_WKHTMLTOPDF_QT_HACK__
 #include "dllbegin.inc"
diff --git a/src/lib/pdf_c_bindings.cc b/src/lib/pdf_c_bindings.cc
index b255395..81daec0 100644
--- a/src/lib/pdf_c_bindings.cc
+++ b/src/lib/pdf_c_bindings.cc
@@ -26,7 +26,7 @@
 #include "pdf_c_bindings_p.hh"
 #include "utilities.hh"
 #include <QApplication>
-#include <QWebFrame>
+#include <qwebframe.h>
 
 #include <QHash>
 
diff --git a/src/lib/pdfconverter.cc b/src/lib/pdfconverter.cc
index 0e4d0c6..8fdf2c6 100644
--- a/src/lib/pdfconverter.cc
+++ b/src/lib/pdfconverter.cc
@@ -27,9 +27,9 @@
 #include <QPair>
 #include <QPrintEngine>
 #include <QTimer>
-#include <QWebFrame>
-#include <QWebPage>
-#include <QWebSettings>
+#include <qwebframe.h>
+#include <qwebpage.h>
+#include <qwebsettings.h>
 #include <QXmlQuery>
 #include <algorithm>
 #include <qapplication.h>
diff --git a/src/lib/pdfconverter_p.hh b/src/lib/pdfconverter_p.hh
index a845afc..cf3a6c6 100644
--- a/src/lib/pdfconverter_p.hh
+++ b/src/lib/pdfconverter_p.hh
@@ -34,10 +34,10 @@
 #include <QPrinter>
 #include <QRegExp>
 #include <QWaitCondition>
-#include <QWebPage>
+#include <qwebpage.h>
 #include <qnetworkreply.h>
 #ifdef __EXTENSIVE_WKHTMLTOPDF_QT_HACK__
-#include <QWebElement>
+#include <qwebelement.h>
 #endif
 
 #include "dllbegin.inc"
diff --git a/src/pdf/pdfdocparts.cc b/src/pdf/pdfdocparts.cc
index 0aef5ea..f70e48b 100644
--- a/src/pdf/pdfdocparts.cc
+++ b/src/pdf/pdfdocparts.cc
@@ -20,7 +20,7 @@
 
 #include "outputter.hh"
 #include "pdfcommandlineparser.hh"
-#include <QWebFrame>
+#include <qwebframe.h>
 
 #define STRINGIZE_(x) #x
 #define STRINGIZE(x) STRINGIZE_(x)
diff --git a/src/pdf/wkhtmltopdf.cc b/src/pdf/wkhtmltopdf.cc
index 297daa4..6bf99fc 100644
--- a/src/pdf/wkhtmltopdf.cc
+++ b/src/pdf/wkhtmltopdf.cc
@@ -23,7 +23,7 @@
 #include <QCommonStyle>
 #include <QPainter>
 #include <QStyleOption>
-#include <QWebFrame>
+#include <qwebframe.h>
 #include <QtPlugin>
 #include <cstdlib>
 #include <iostream>
