diff --git a/voltron/core.py b/voltron/core.py
index f22f375..351a0ea 100644
--- a/voltron/core.py
+++ b/voltron/core.py
@@ -15,7 +15,8 @@ import six
 import voltron
 from flask import Flask, Response, make_response, redirect, render_template, request
 from werkzeug.serving import BaseWSGIServer, ThreadedWSGIServer, WSGIRequestHandler
-from werkzeug.wsgi import DispatcherMiddleware, SharedDataMiddleware
+from werkzeug.middleware.dispatcher import DispatcherMiddleware
+from werkzeug.middleware.shared_data import SharedDataMiddleware
 from requests import ConnectionError
 
 
